Core/include/Acts/Definitions/Algebra.hpp

```diff
@@ -22,71 +22,75 @@
+/// meaning in interfaces. It also ensures that the matrix is square. However,
+/// the user is responsible for ensuring that the values are symmetric.
+///
+/// Without a distinct type for symmetric matrices, there is no way to provide
+/// any conditions e.g. square size, for the dynamic-sized case. Consequently,
+/// no dynamic-sized symmetric matrix type is defined. Use the
+/// `ActsDynamicMatrix` instead.
+///
+/// @{
+
+/// Common scalar (floating point type used for the default algebra types.
+///
+/// Defaults to `double` but can be customized by the user.
 #ifdef ACTS_CUSTOM_SCALARTYPE
 using ActsScalar = ACTS_CUSTOM_SCALARTYPE;
 #else
 using ActsScalar = double;
 #endif
 
-template <typename T, unsigned int rows, unsigned int cols>
-using ActsMatrix = Eigen::Matrix<T, rows, cols>;
+template <unsigned int kSize>
+using ActsVector = Eigen::Matrix<ActsScalar, kSize, 1>;
 
-template <unsigned int rows, unsigned int cols>
-using ActsMatrixD = ActsMatrix<ActsScalar, rows, cols>;
+template <unsigned int kRows, unsigned int kCols>
+using ActsMatrix = Eigen::Matrix<ActsScalar, kRows, kCols>;
 
-template <typename T, unsigned int rows>
-using ActsSymMatrix = Eigen::Matrix<T, rows, rows>;
+template <unsigned int kSize>
+using ActsSymMatrix = Eigen::Matrix<ActsScalar, kSize, kSize>;
 
-template <unsigned int rows>
-using ActsSymMatrixD = ActsSymMatrix<ActsScalar, rows>;
+using ActsDynamicVector = Eigen::Matrix<ActsScalar, Eigen::Dynamic, 1>;
 
-template <typename T, unsigned int rows>
-using ActsVector = Eigen::Matrix<T, rows, 1>;
+using ActsDynamicMatrix =
+    Eigen::Matrix<ActsScalar, Eigen::Dynamic, Eigen::Dynamic>;
 
-template <unsigned int rows>
-using ActsVectorD = ActsVector<ActsScalar, rows>;
+/// @}
 
-template <typename T, unsigned int cols>
-using ActsRowVector = Eigen::Matrix<T, 1, cols>;
-
-template <unsigned int cols>
-using ActsRowVectorD = ActsRowVector<ActsScalar, cols>;
-
-template <typename T>
-using ActsMatrixX = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>;
-
-using ActsMatrixXd = ActsMatrixX<ActsScalar>;
-
-template <typename T>
-using ActsVectorX = Eigen::Matrix<T, Eigen::Dynamic, 1>;
-
-/// @defgroup fixed-algebra-types Fixed-size vector/matrix e.g. for coordinates
+/// @defgroup coordinates-types Fixed-size vector/matrix types for coordinates
 ///
-/// These predefined types should always be used when handling the coordinate
-/// vectors in different coordinate systems, i.e. local (2d), global spatial
-/// (3d), or global space-time (4d).
+/// These predefined types should always be used when handling coordinate
+/// vectors in different coordinate systems, i.e. on surfaces (2d), spatial
+/// position (3d), or space-time (4d).
 ///
 /// @{
 
 // coordinate vectors
-using Vector2D = ActsVector<ActsScalar, 2>;
-using Vector3D = ActsVector<ActsScalar, 3>;
-using Vector4D = ActsVector<ActsScalar, 4>;
+using Vector2D = ActsVector<2>;
+using Vector3D = ActsVector<3>;
+using Vector4D = ActsVector<4>;
```

Comment:
So these are just aliases?

Reply:
Yes.

Reply:
Then I don't quite understand why we don't just stick to the names we use already.

Reply:
I do not quite understand. The `Vector{2,3,4}D` et al. types are the ones that we currently use?

Reply:
Hm. I guess this is what you meant renaming `VectorXD` to `ActsVectorX`. Ok nevermind, let's leave this as is.

Reply:
The PR description was not clear enough: this only streamlines the generic vector/matrix types with configurable size, not the fixed-sized types intended for coordinates.